health_monitoring/test_health_monitor.py

```diff
@@ -3,235 +3,193 @@
-                'request_url "/page.html";\n',
-                'timeout 15;\n'
-            ]
-            + [crc32_conf] + [rcodes_conf] + ['}\n']
-        )
-        rules_config = ''.join(
-            ['http_chain {\n'] +
-            ['-> %s;\n' % vhost] +
-            ['}\n']
-        )
-        return config + hm_config + rules_config
-
-    def create_tester(self):
-        self.tester = StagedDeproxy(self.client, self.servers)
-        self.tester.configure(self.tempesta, self.srv_group, [
-            Stage(self.tester, self.client, self.ch_enabled, True),
-            Stage(self.tester, self.client, self.ch_disabled, False),
-            Stage(self.tester, self.client, self.ch_enabled, True,
-                  trans=self.trans_resp),
-            Stage(self.tester, self.client, self.ch_disabled, False),
-            Stage(self.tester, self.client, self.ch_enabled, True,
-                  trans=self.trans_resp)
-        ])
-
-    def create_servers(self):
-        p = tempesta.upstream_port_start_from()
-        self.servers = [deproxy.Server(port=p, conns_n=3)]
-
-    def configure_tempesta(self):
-        sg = tempesta.ServerGroup(self.srv_group, hm='h_monitor1')
-        for s in self.servers:
-            sg.add_server(s.ip, s.port, s.conns_n)
-        self.tempesta.config.add_sg(sg)
-
-    def assert_tempesta(self):
-        self.tester.assert_stages()
-        err_msg = 'Tempesta must have errors due' \
-                  ' to inability to schedule request'
-        functional.FunctionalTest.assert_tempesta(self)
-        self.assertTrue(self.tempesta.stats.cl_msg_other_errors > 0,
-                        msg=err_msg)
+    tempesta = {
+        'config': TEMPESTA_CONFIG % "",
+    }
+
+    backends = [
+        {
+            'id' : 'nginx1',
+            'type' : 'nginx',
+            'port' : '8080',
+            'status_uri' : 'http://${server_ip}:8080/nginx_status',
+            'config' : NGINX_CONFIG % (8080, """
+
+return 403;
+"""),
+        },{
+            'id' : 'nginx2',
+            'type' : 'nginx',
+            'port' : '8081',
+            'status_uri' : 'http://${server_ip}:8081/nginx_status',
+            'config' : NGINX_CONFIG % (8081, """
+
+return 404;
+"""),
+        },{
+            'id' : 'nginx3',
+            'type' : 'nginx',
+            'port' : '8082',
+            'status_uri' : 'http://${server_ip}:8082/nginx_status',
+            'config' : NGINX_CONFIG % (8082, """
+
+return 200;
+"""),        
+        }
+    ]
+
+    clients = [
+        {
+            'id': 'curl',
+            'type': 'external',
+            'binary': 'curl',
+            'cmd_args': (
+                    ' -o /dev/null -s -w "%{http_code}\n" '
+                    'http://${tempesta_ip}'
+            )
+        },
+    ]
+
+
+    def wait_for_server(self, srv):
+        srv.start()
+        while srv.state != 'started':
+            pass
+        srv.wait_for_connections()
+
```

Comment:
Only one 1 line should be used according to https://peps.python.org/pep-0008/#blank-lines . We do need to merge https://github.com/tempesta-tech/tempesta-test/pull/241 or other linter...